File: t/CS/GrayScaleDemo/UI/EdgeDetectorDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Leadtools.Demos;
using Leadtools.ImageProcessing.Effects;

namespace GrayScaleDemo
{
    public partial class EdgeDetectorDialog : Form
    {
        private static int _initialThreshold = 255;
        private static EdgeDetectorCommandType _initialFilter = EdgeDetectorCommandType.SobelVertical;
        private int _max,_min;

        public int Threshold;
        public EdgeDetectorCommandType Filter;

        public EdgeDetectorDialog(int min,int max,bool isGray)
        {
            InitializeComponent();

            if (isGray)
            {
               _max = max;
               _min = min;
            }
            else
            {
               _max = 255;
               _min = 0;
            }
        }

        private void EdgeDetectorDialog_Load(object sender, EventArgs e)
        {
             EdgeDetectorCommand command = new EdgeDetectorCommand();
             _initialThreshold = command.Threshold;
             _initialFilter = command.Filter;


            Threshold = _initialThreshold;
            Filter = _initialFilter;

            Tools.FillComboBoxWithEnum(_cbFilter, typeof(EdgeDetectorCommandType), Filter);

            _numThreshold.Maximum = _max;
            _numThreshold.Minimum = _min;
            DialogUtilities.SetNumericValue(_numThreshold, Threshold);
        }

        private void _btnOk_Click(object sender, EventArgs e)
        {
            Threshold = (int)_numThreshold.Value;

            Filter = (EdgeDetectorCommandType)Constants.GetValueFromName(
               typeof(EdgeDetectorCommandType),
               (string)_cbFilter.SelectedItem,
               _initialFilter);

            _initialThreshold = Threshold;
            _initialFilter = Filter;
        }
    }
}

File: t/VB/GrayScaleDemo/UI/EdgeDetectorDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************

Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports GrayScaleDemo.Leadtools.Demos
Imports Leadtools.ImageProcessing.Effects
Imports System

Partial Public Class EdgeDetectorDialog
   Inherits Form
   Private Shared _initialThreshold As Integer = 255
   Private Shared _initialFilter As EdgeDetectorCommandType = EdgeDetectorCommandType.SobelVertical
   Private _max As Integer, _min As Integer

   Public Threshold As Integer
   Public Filter As EdgeDetectorCommandType

   Public Sub New(min As Integer, max As Integer, isGray As Boolean)
      InitializeComponent()

      If isGray Then
         _max = max
         _min = min
      Else
         _max = 255
         _min = 0
      End If
   End Sub

   Private Sub EdgeDetectorDialog_Load(sender As Object, e As EventArgs) Handles Me.Load
      Dim command As New EdgeDetectorCommand()
      _initialThreshold = command.Threshold
      _initialFilter = command.Filter


      Threshold = _initialThreshold
      Filter = _initialFilter

      Tools.FillComboBoxWithEnum(_cbFilter, GetType(EdgeDetectorCommandType), Filter)

      _numThreshold.Maximum = _max
      _numThreshold.Minimum = _min
      DialogUtilities.SetNumericValue(_numThreshold, Threshold)
   End Sub

   Private Sub _btnOk_Click(sender As Object, e As EventArgs) Handles _btnOk.Click
      Threshold = CInt(_numThreshold.Value)

      Filter = CType(Leadtools.Demos.Constants.GetValueFromName(GetType(EdgeDetectorCommandType), CType(_cbFilter.SelectedItem, String), _initialFilter), EdgeDetectorCommandType)

      _initialThreshold = Threshold
      _initialFilter = Filter
   End Sub
End Class
